arboretum/ibm_cloud/fetchers/fetch_cluster_resource.py

```diff
@@ -0,0 +1,165 @@
+                    'auth-provider']['config']['id-token']
+            if p.name.endswith('.pem'):
+                cluster_configs[cluster['name']].extract(
+                    name, path=self.tempdir.name
+                )
+                ca_cert_filepath = os.path.join(self.tempdir.name, name)
+        return cluster_token, ca_cert_filepath
+
+    def _get_roks_credentials(self, cluster, api_key):
+        """Get credentials for a ROKS cluster.
+
+        This function implements the procedure described in
+        https://cloud.ibm.com/docs/openshift?topic=openshift-access_cluster#access_automation
+        """
+        s = self.session(cluster['serverURL'])
+        oauth_path = '/.well-known/oauth-authorization-server'
+        resp = s.get(oauth_path)
+        resp.raise_for_status()
+        token_endpoint = resp.json()['token_endpoint']
+        oauth_server = token_endpoint.split('/')[2]
+        s = self.session(f'https://{oauth_server}')
+        token_path = (
+            '/oauth/authorize?client_id='
+            'openshift-challenging-client&response_type=token'
+        )
+        resp = s.get(
+            token_path,
+            auth=('apikey', api_key),
+            headers={'X-CSRF-Token': 'a'},
+            allow_redirects=False
+        )
+        location = resp.headers['Location']
+        keyword = 'access_token='
+        start = location.find(keyword)
+        end = location.find('&', start)
+        cluster_token = location[start + len(keyword):end]
+        s.headers.update = ({'X-CSRF-Token': None})
+
+        return cluster_token, None
+
+    @store_raw_evidence('ibm_cloud/cluster_resources.json')
+    def fetch_cluster_resource(self):
+        """Fetch cluster resources."""
+        cluster_list_evidence = get_evidence_dependency(
+            'raw/ibm_cloud/cluster_list.json', self.locker
+        )
+        cluster_list = json.loads(cluster_list_evidence.content)
+        resources = {}
+        for account in cluster_list:
+            api_key = getattr(
+                self.config.creds['ibm_cloud'], f'{account}_api_key'
+            )
+            headers = {'Accept': 'application/json'}
+            self.session('https://containers.cloud.ibm.com', **headers)
+            access_token, refresh_token = get_tokens(api_key)
+            self.session().headers.update(
+                {
+                    'Authorization': 'Bearer '
+                    f'{access_token}',
+                    'X-Auth-Refresh-Token': refresh_token
+                }
+            )
+            cluster_config = {}
+            for cluster in cluster_list[account]:
+                resp = self.session().get(
+                    '/global/v1/clusters/'
+                    f'{cluster["id"]}/config'
+                )
+                resp.raise_for_status()
+                cluster_config[cluster['name']] = zipfile.ZipFile(
+                    io.BytesIO(resp.content)
+                )
```

Comment:
What is the purpose of compressing the content here and attaching that content as a value in this dictionary?

Reply:
This line just create a ZipFile object with zipfile bytestream (`io.BytesIO(resp.content)`) given by the server, and does not perform compression/decompression. The zipfile object is extracted in [_get_iks_credentials()](https://github.com/ComplianceAsCode/auditree-arboretum/pull/54/files#diff-bcd498e6b9c862fab4f20772411039245ac766ae4cab4186d5c9285b8835064aR76).